Samples/WinMLSamplesGallery/TransformAsync/SegmentModel.cpp

```diff
@@ -0,0 +1,279 @@
+			.SetAttribute(L"keepdims", TensorInt64Bit::CreateFromArray({ 1 }, { 1 }))
+			.SetAttribute(L"axis", TensorInt64Bit::CreateFromIterable({ 1 }, { axis })) // Correct way of passing axis? 
+			.SetOutput(L"reduced", L"Reduced"))
+		.Operators().Add(LearningModelOperator(L"Cast")
+			.SetInput(L"input", L"Reduced")
+			.SetAttribute(L"to", TensorInt64Bit::CreateFromIterable({}, { OnnxDataType::ONNX_FLOAT }))
+			.SetOutput(L"output", L"ArgmaxOutput"))
+
+		// Extract the foreground using the argmax scores to create a mask
+		.Operators().Add(LearningModelOperator(L"Clip")
+			.SetInput(L"input", L"ArgmaxOutput")
+			.SetConstant(L"min", TensorFloat::CreateFromIterable({ 1 }, { 0.f }))
+			.SetConstant(L"max", TensorFloat::CreateFromIterable({ 1 }, { 1.f }))
+			.SetOutput(L"output", L"MaskBinary"))
+		.Operators().Add(LearningModelOperator(L"Mul")
+			.SetInput(L"A", L"InputImage")
+			.SetInput(L"B", L"MaskBinary")
+			.SetOutput(L"C", L"ForegroundImage"))
+
+		// Extract the blurred background using the negation of the foreground mask
+		.Operators().Add(LearningModelOperator(L"AveragePool") // AveragePool to create blurred background
+			.SetInput(L"X", L"InputImage")
+			.SetAttribute(L"kernel_shape", TensorInt64Bit::CreateFromArray(std::vector<int64_t>{2}, std::array<int64_t, 2>{20, 20}))
+			.SetAttribute(L"auto_pad", TensorString::CreateFromArray(std::vector<int64_t>{1}, std::array<hstring, 1>{L"SAME_UPPER"}))
+			.SetOutput(L"Y", L"BlurredImage"))
+		.Operators().Add(LearningModelOperator(L"Mul")
+			.SetInput(L"A", L"MaskBinary")
+			.SetConstant(L"B", TensorFloat::CreateFromIterable({ 1 }, { -1.f }))
+			.SetOutput(L"C", L"NegMask"))
+		.Operators().Add(LearningModelOperator(L"Add") // BackgroundMask = (1- foreground Mask)
+			.SetConstant(L"A", TensorFloat::CreateFromIterable({ 1 }, { 1.f }))
+			.SetInput(L"B", L"NegMask")
+			.SetOutput(L"C", L"BackgroundMask"))
+		.Operators().Add(LearningModelOperator(L"Mul") // Extract the blurred background
+			.SetInput(L"A", L"BlurredImage")
+			.SetInput(L"B", L"BackgroundMask")
+			.SetOutput(L"C", L"BackgroundImage"))
+
+		// Combine foreground and background
+		.Operators().Add(LearningModelOperator(L"Add")
+			.SetInput(L"A", L"ForegroundImage")
+			.SetInput(L"B", L"BackgroundImage")
+			.SetOutput(L"C", L"OutputImage"))
+		;
+
+	return builder.CreateModel();
+}
+
+LearningModel Invert(long n, long c, long h, long w)
+{
+	auto builder = LearningModelBuilder::Create(opset)
+		// Loading in buffers and reshape
+		.Inputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Input", TensorKind::Float, { n, c, h, w }))
+		.Outputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Output", TensorKind::Float, { n, c, h, w }))
+		.Operators().Add(LearningModelOperator(L"Mul")
+			.SetInput(L"A", L"Input")
+			.SetConstant(L"B", TensorFloat::CreateFromIterable({ 1 }, { -1.f }))
+			//.SetConstant(L"B", TensorFloat::CreateFromIterable({3}, {0.114f, 0.587f, 0.299f}))
```

Comment:
>//.SetConstant(L"B", TensorFloat::CreateFromIterable({3}, {0.114f, 0.587f, 0.299f})) [](http://example.com/codeflow?start=3&length=84)

I'm guessing based on the values that these are B G R weights?

We should either delete the comment or add more information, so the user isn't confused.

Reply:
deleted